packages/core/rendering/core-rendering-server-internal/src/rendering_service.tsx

```diff
@@ -147,8 +147,21 @@ export class RenderingService {
       globalSettingsUserValues = userValues[1];
     }
 
+    const defaultSettings = uiSettings.client?.getRegistered() ?? {};
+
+    // Load async settings values
+    await Promise.all(
+      Object.entries(defaultSettings)
+        .filter(([_, definition]) => typeof definition.getValue === 'function')
+        .map(([key, definition]) => {
+          return definition.getValue!({ request }).then((value) => {
+            defaultSettings[key].value = value;
+          });
+        })
+    );
```

Comment:
NIT: can we please extract this to a pretty utility function (in the same file is fine)

Reply:
Sure will do 👍

Reply:
Addressed in https://github.com/elastic/kibana/pull/187763/commits/ebdd1f4c026a6d0690697181a31b68e4620977f8